tests/test_configmanager.py

```diff
@@ -159,33 +161,28 @@ def test_parse(self):
         self.assertIn('he-ipv6', self.configmanager.virtual_interfaces)
 
     def test_parse_merging(self):
-        self.configmanager.parse(extra_config=[os.path.join(self.workdir.name, "newfile_merging.yaml")])
-        self.assertIn('eth0', self.configmanager.ethernets)
-        self.assertIn('dhcp4', self.configmanager.ethernets['eth0'])
-        self.assertEquals(True, self.configmanager.ethernets['eth0'].get('dhcp6'))
-        self.assertEquals(True, self.configmanager.ethernets['ethbr1'].get('dhcp4'))
+        state = self.configmanager.parse(extra_config=[os.path.join(self.workdir.name, "newfile_merging.yaml")])
+        self.assertIn('eth0',   state.ethernets)
+        self.assertIn('eth42',   state.ethernets)
 
     def test_parse_merging_ovs(self):
-        self.configmanager.parse(extra_config=[os.path.join(self.workdir.name, "ovs_merging.yaml")])
-        self.assertIn('eth0', self.configmanager.ethernets)
-        self.assertIn('dhcp4', self.configmanager.ethernets['eth0'])
-        self.assertIn('patchx', self.configmanager.ovs_ports)
-        self.assertIn('patchy', self.configmanager.ovs_ports)
-        self.assertIn('ovs0', self.configmanager.bridges)
-        self.assertEqual({}, self.configmanager.ovs_ports['patchx'].get('openvswitch'))
-        self.assertEqual({}, self.configmanager.ovs_ports['patchy'].get('openvswitch'))
-        self.assertEqual({}, self.configmanager.bridges['ovs0'].get('openvswitch'))
+        state = self.configmanager.parse(extra_config=[os.path.join(self.workdir.name, "ovs_merging.yaml")])
+        self.assertIn('eth0',   state.ethernets)
+        # self.assertIn('dhcp4',  state.ethernets['eth0'])
+        self.assertIn('patchx', state.ovs_ports)
+        self.assertIn('patchy', state.ovs_ports)
+        self.assertIn('ovs0',   state.bridges)
+        self.assertEqual('OpenVSwitch', state['ovs0'].backend)
```

Comment:
I think we should be able to do the same `backend == 'OpenVSwitch'` check for patchx & patchy, are we?

Reply:
Tests added.